gc/base/SparseVirtualMemory.cpp

```diff
@@ -163,22 +172,50 @@ MM_SparseVirtualMemory::freeSparseRegionAndUnmapFromHeapObject(MM_EnvironmentBas
 	bool ret = true;
 
 	if ((NULL != dataPtr) && (0 != dataSize)) {
-		Assert_MM_true(0 == (dataSize % _pageSize));
-		ret = decommitMemory(env, dataPtr, dataSize);
+//		Assert_MM_true(0 == (dataSize % _pageSize));
+		uintptr_t adjustedSize = adjustSize(dataSize);
+//		ret = decommitMemory(env, dataPtr, dataSize);
+		ret = decommitMemory(env, dataPtr, adjustedSize);
+		if (ret) {
+			omrthread_monitor_enter(_largeObjectVirtualMemoryMutex);
+//			ret = (_sparseDataPool->returnFreeListEntry(dataPtr, dataSize) && _sparseDataPool->unmapSparseDataPtrFromHeapProxyObjectPtr(dataPtr));
+			ret = (_sparseDataPool->returnFreeListEntry(dataPtr, adjustedSize) && _sparseDataPool->unmapSparseDataPtrFromHeapProxyObjectPtr(dataPtr));
+			omrthread_monitor_exit(_largeObjectVirtualMemoryMutex);
+			Trc_MM_SparseVirtualMemory_decommitMemory_success(dataPtr, (void *)adjustedSize);
+		} else {
+			Trc_MM_SparseVirtualMemory_decommitMemory_failure(dataPtr, (void *)adjustedSize);
+			/* TODO: Assert Fatal in case of failure? */
```

Comment:
yes, I think this should be converted to a fatal failure (can use 'unreachable' assert)

again, opinion from @dmitripivkine ?

Reply:
We can use assert. There are only few corner cases when this call can fail. For most platforms underlying `omrvmem_decommit_memory()` can fail by misaligned parameters only. But `MM_VirtualMemory` layer should care about parameters alignment.